### PR TITLE
Major: Rewrite to support SQL

### PR DESCRIPTION
**Pull request by Rushaway**
- Now support SQL
- Cleanned code
- Include file

Co-Authored-By: Dolly132 <109222243+Dolly132@users.noreply.github.com>

### CODE REVIEW

File: addons/sourcemod/scripting/SelfMute.sp [hunk 1]
```diff
@@ -75,6 +118,20 @@ public void OnPluginStart()
 		SetFailState("This game doesn't support SendAudio user messages.");
 
 	HookUserMessage(SendAudio, Hook_UserMessageSendAudio, true);
+
+	ConnectToDB();
+	
+	for(int i = 1; i <= MaxClients; i++)
+	{
+        if(IsClientInGame(i))
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 2]
```diff
@@ -41,25 +66,43 @@ enum
...
 
+	g_hCVar_Debug = CreateConVar("sm_selfmute_debug_level", "1", "[0 = Disabled | 1 = Error | 2 = Extended]", FCVAR_REPLICATED);
+	AutoExecConfig(true);
+
+	HookEvent("player_team", Event_TeamChange);
 	HookEvent("round_start", Event_Round);
 	HookEvent("round_end", Event_Round);
-	HookEvent("player_team", Event_TeamChange);
+	
+	g_hSmModeCookie = RegClientCookie("SmMode_Cookie", "Selfmute mode cookie", CookieAccess_Public);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: selfmute_mode

File: addons/sourcemod/scripting/SelfMute.sp [hunk 3]
```diff
@@ -41,25 +66,43 @@ enum
...
+	g_hBotSmCookie = RegClientCookie("SmBot_Cookie", "Selfmute bot", CookieAccess_Public);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: selfmute_bot

File: addons/sourcemod/scripting/SelfMute.sp [hunk 4]
```diff
@@ -41,25 +66,43 @@ enum
...
-	CreateConVar("sm_selfmute_version", PLUGIN_VERSION, "Version of Self-Mute", FCVAR_SPONLY|FCVAR_REPLICATED|FCVAR_NOTIFY);
-
 	RegConsoleCmd("sm_sm", Command_SelfMute, "Mute player by typing !sm [playername]");
 	RegConsoleCmd("sm_su", Command_SelfUnMute, "Unmute player by typing !su [playername]");
 	RegConsoleCmd("sm_cm", Command_CheckMutes, "Check who you have self-muted");
+	RegConsoleCmd("sm_suall", Command_SelfUnMuteAll, "Unmute all clients/groups");
+	RegConsoleCmd("sm_smcookies", Command_SmCookies, "Choose the good cookie");
+	SetCookieMenuItem(CookieMenu_Handler, 0, "SelfMute Cookies");
 
+	g_hCVar_Debug = CreateConVar("sm_selfmute_debug_level", "1", "[0 = Disabled | 1 = Error | 2 = Extended]", FCVAR_REPLICATED);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: sm_selfmute_log_level

2 = Info

File: addons/sourcemod/scripting/SelfMute.sp [hunk 5]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+}
+
+public Action OnClientJoinCheck(Handle timer, int userid)
+{   
+	int client = GetClientOfUserId(userid);	
+	if(client < 1)
+		return Plugin_Stop;
+		
+	if(!IsClientInGame(client))
+        return Plugin_Stop;
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 6]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
...
+		CPrintToChat(client, "%sCould not read your friendslist, your profile must be set to public!", PLUGIN_PREFIX);
 		SpecialMute &= ~MUTE_NOTFRIENDS;
 		RetValue = true;
 	}
-
+	
 	if(SpecialMute)
 	{
-		if(SpecialMute & MUTE_ALL || g_SpecialMutes[client] & MUTE_ALL)
+		if(g_SpecialMutes[client] & SpecialMute)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 7]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
...
+	    {
+	        CPrintToChat(client, "%sYou have already self-muted this group.", PLUGIN_PREFIX);
+	        return true;
+	    }
+	
+		else if(g_SpecialMutes[client] == MUTE_ALL)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 8]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
...
+	    {
+	        CPrintToChat(client, "%sYou have muted everyone, so how do you want to mute any other group?", PLUGIN_PREFIX);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: You have muted everyone, do you want to mute another group?

File: addons/sourcemod/scripting/SelfMute.sp [hunk 9]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
...
 		char aBuf[128];
 		FormatSpecialMutes(SpecialMute, aBuf, sizeof(aBuf));
-
-		PrintToChat(client, "\x04[Self-Mute]\x01 You have self-muted group:\x04 %s", aBuf);
+		UpdateSpecialMutesThisClient(client);
+		
+		if(g_bClientJustJoined[client])
+			return true;
+	
+		switch(g_iClientSmMode[client])
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 10]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
...
+	                {
+	                    DeleteGroupFromTable(client, SteamID, Argument);
+	                    InsertGroupToTable(client, SteamID, aBuf, Argument);
+	                    UpdateSpecialMutesThisClient(client);
+	                    if(IsClientInGame(client))
+	                        CPrintToChat(client, "%sYou have self-muted {olive}%s{default}. (Permanently)", PLUGIN_PREFIX, aBuf);
+	                }
+	            }
+	        }
+			case SmMode_Alert:
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation of whole block

File: addons/sourcemod/scripting/SelfMute.sp [hunk 11]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+	if(IsFakeClient(client))
+		return;
+	
+	CreateTimer(2.0, OnClientJoinCheck, GetClientUserId(client));
+}
+
+public Action OnClientJoinCheck(Handle timer, int userid)
+{   
+	int client = GetClientOfUserId(userid);	
+	if(client < 1)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: || client > MaxClients

File: addons/sourcemod/scripting/SelfMute.sp [hunk 12]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+	T_ClientJoin.AddQuery(sQuery2);
+	g_hDB.Execute(T_ClientJoin, SQL_OnClientJoinSuccess, SQL_OnClientJoinError, userid);
+	
+	return Plugin_Continue;
+}
+
+public void SQL_OnClientJoinSuccess(Database hDatabase, int userid, int iNumQueries, Handle[] hResults, any[] QueryData)
+{
+	int client = GetClientOfUserId(userid);
+	if(client < 1)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: || client > MaxClients

File: addons/sourcemod/scripting/SelfMute.sp [hunk 13]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+{
+	if(result == null)
+	{
+		if (g_hCVar_Debug.IntValue >= 1)
+			LogError("[Self-Mute] Error while deleting all client mutes from table. error: %s", error);
+		return;
+	}
+	
+	int client = GetClientOfUserId(userid);
+	if(client < 1) // player disconnected
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: || client > MaxClients

File: addons/sourcemod/scripting/SelfMute.sp [hunk 14]
```diff
@@ -413,14 +1052,78 @@ bool UnMuteSpecial(int client, char[] Argument)
 
 void Ignore(int client, int target)
 {
+	if(client < 1 || target < 1)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: || client > MaxClients || target > MaxClients

File: addons/sourcemod/scripting/SelfMute.sp [hunk 15]
```diff
@@ -413,14 +1052,78 @@ bool UnMuteSpecial(int client, char[] Argument)
...
+
+	g_bClientTargets[client][target] = true;
 	SetIgnored(client, target, true);
+	UpdateIgnored();
 	SetListenOverride(client, target, Listen_No);
 }
 
 void UnIgnore(int client, int target)
 {
+	if(client < 1 || target < 1)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: || client > MaxClients || target > MaxClients

File: addons/sourcemod/scripting/SelfMute.sp [hunk 16]
```diff
@@ -624,30 +1391,710 @@ public Action Command_CheckMutes(int client, int args)
...
+	}
+}
+
+public void SQL_AddMutesToMenu(Handle hDataBase, Handle query, const char[] error, int userid)
+{
+	if(query == null)
+		return;
+
+	int client = GetClientOfUserId(userid);
+	if(client < 1)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: || client > MaxClients

File: addons/sourcemod/scripting/SelfMute.sp [hunk 17]
```diff
@@ -624,30 +1391,710 @@ public Action Command_CheckMutes(int client, int args)
...
+			g_hDB.Format(sQuery, sizeof(sQuery), "SELECT `group_filter`, `group_name` FROM `GroupsMute` WHERE client_steamid='%s'", SteamID);
+			SQL_TQuery(g_hDB, SQL_GroupsMenu, sQuery, userid);
+		}
+	}
+}
+
+public void SQL_GroupsMenu(Handle hDatabase, Handle query, const char[] error, int userid)
+{
+	int client = GetClientOfUserId(userid);
+	if(client < 1)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: || client > MaxClients

File: addons/sourcemod/scripting/SelfMute.sp [hunk 18]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+	if(g_hDB == null)
+		return;
+	
+	char ClientSteamID[32], TargetSteamID[32], ClientName[64], TargetName[64];
+	
+	if(!GetClientName(client, ClientName, sizeof(ClientName)) || !GetClientName(target, TargetName, sizeof(TargetName)))
+		return;
+	
+	if(!GetClientAuthId(client, AuthId_Steam2, ClientSteamID, sizeof(ClientSteamID)) || !GetClientAuthId(target, AuthId_Steam2, TargetSteamID, sizeof(TargetSteamID)))
+        return;
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 19]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+
+	g_bClientJustJoined[client] = false;
+	UpdateIgnored();
+}
+
+public void SQL_OnClientJoinError(Database hDatabase, any Data, int iNumQueries, const  char[] sError, int iFailIndex, any[] QueryData)
+{
+	if (g_hCVar_Debug.IntValue >= 1)
+		LogError("[Self-Mute] Error while getting client data on connect, error: %s", sError);
+	return;
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: no need its a void function

File: addons/sourcemod/scripting/SelfMute.sp [hunk 20]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+	}
+	
+	g_hDB.Execute(T_Delete, DB_DeleteOnSuccess, DB_DeleteOnError);
+}
+
+public void DB_DeleteOnSuccess(Database hDatabase, any data, int iNumQueries, Handle[] hResults, any[] QueryData)
+{
+	if (g_hCVar_Debug.IntValue >= 2)
+		LogMessage("[Self-Mute] Successfully deleted data from database");
+	return;
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: no need its a void function

File: addons/sourcemod/scripting/SelfMute.sp [hunk 21]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+}
+
+public void DB_DeleteOnError(Database hDatabase, any Data, int iNumQueries, const  char[] sError, int iFailIndex, any[] QueryData)
+{
+	if (g_hCVar_Debug.IntValue >= 1)
+		LogError("[Self-Mute] Error while deleting data from database, error: %s", sError);
+	return;
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: no need its a void function

File: addons/sourcemod/scripting/SelfMute.sp [hunk 22]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+        g_iClientSmMode[client] = StringToInt(sValue);
+    else
+        g_iClientSmMode[client] = SmMode_Alert; 
+        
+    g_bClientJustJoined[client] = true;
+    if(StrEqual(sBool, "1"))
+    {
+        for(int i = 1; i <= MaxClients; i++)
+        {
+			if(!IsClientInGame(i))
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 23]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+				continue;
+				
+			if(IsClientSourceTV(i))
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 24]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
 {
 	bool RetValue = false;
 	int SpecialMute = GetSpecialMutesFlags(Argument);
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 25]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
...
+	
 	if(SpecialMute & MUTE_NOTFRIENDS && g_Plugin_AdvancedTargeting && ReadClientFriends(client) != 1)
 	{
-		PrintToChat(client, "\x04[Self-Mute]\x01 Could not read your friendslist, your profile must be set to public!");
+		CPrintToChat(client, "%sCould not read your friendslist, your profile must be set to public!", PLUGIN_PREFIX);
 		SpecialMute &= ~MUTE_NOTFRIENDS;
 		RetValue = true;
 	}
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 26]
```diff
@@ -341,33 +890,102 @@ bool MuteSpecial(int client, char[] Argument)
...
+	        return true;
+	    }
+	
+		else if(g_SpecialMutes[client] == MUTE_ALL)
+	    {
+	        CPrintToChat(client, "%sYou have muted everyone, so how do you want to mute any other group?", PLUGIN_PREFIX);
+	        return true;
+	    }
+	    
+		if(g_iClientSmMode[client] != SmMode_Alert || g_bClientJustJoined[client])
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation of whole block

File: addons/sourcemod/scripting/SelfMute.sp [hunk 27]
```diff
@@ -413,14 +1052,78 @@ bool UnMuteSpecial(int client, char[] Argument)
...
+
 	SetIgnored(client, target, false);
+	UpdateIgnored();
 	SetListenOverride(client, target, Listen_Yes);
+	g_bClientSavedTargets[client][target] = false;
+	g_bClientNotSavedTargets[client][target] = false;
+	g_bClientTargets[client][target] = false;
+	
+	if(IsClientSourceTV(target))
+    {
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 28]
```diff
@@ -413,14 +1052,78 @@ bool UnMuteSpecial(int client, char[] Argument)
...
+		SetClientCookie(client, g_hBotSmCookie, "0");
+		return;
+    }
+	char SteamID[32];
+	if(!GetClientAuthId(client, AuthId_Steam2, SteamID, sizeof(SteamID)) || !GetClientAuthId(target, AuthId_Steam2, SteamID, sizeof(SteamID)))
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp:1118]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 29]
```diff
@@ -486,30 +1189,48 @@ public Action Command_SelfMute(int client, int args)
...
 		{
 			UnExempt(client, aTargetList[0]);
 
-			PrintToChat(client, "\x04[Self-Mute]\x01 You have removed exempt from self-mute:\x04 %s", sTargetName);
+			CReplyToCommand(client, "%sYou have removed exempt from self-mute{olive} %s{default}.", PLUGIN_PREFIX, sTargetName);
 
 			return Plugin_Handled;
 		}
+
+		if(g_bClientTargets[client][aTargetList[0]])
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp:1200]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 30]
```diff
@@ -486,30 +1189,48 @@ public Action Command_SelfMute(int client, int args)
...
+        {
+            CReplyToCommand(client, "%sYou have already self-muted {olive}%N{default}.", PLUGIN_PREFIX, aTargetList[0]);
+            return Plugin_Handled;
+        }
 	}
+	else if(TargetCount > 1)
+    {
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: Indentation

File: addons/sourcemod/scripting/SelfMute.sp [hunk 31]
```diff
@@ -518,32 +1239,32 @@ public Action Command_SelfUnMute(int client, int args)
 {
 	if(client == 0)
 	{
-		PrintToServer("[SM] Cannot use command from server console.");
+		ReplyToCommand(client, "[SM] Cannot use command from server console.");
 		return Plugin_Handled;
 	}
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 32]
```diff
@@ -518,32 +1239,32 @@ public Action Command_SelfUnMute(int client, int args)
...
-
+	
 	if(args < 1)
 	{
-		DisplayUnMuteMenu(client);
+		DisplayClientMutesMenu(client);
 		return Plugin_Handled;
 	}
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 33]
```diff
@@ -518,32 +1239,32 @@ public Action Command_SelfUnMute(int client, int args)
...
-		DisplayUnMuteMenu(client);
+		DisplayClientMutesMenu(client);
 		return Plugin_Handled;
 	}
-
+	
 	char Argument[65];
 	GetCmdArg(1, Argument, sizeof(Argument));
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 34]
```diff
@@ -518,32 +1239,32 @@ public Action Command_SelfUnMute(int client, int args)
...
 	char Argument[65];
 	GetCmdArg(1, Argument, sizeof(Argument));
-
+	
 	char Filtered[65];
 	strcopy(Filtered, sizeof(Filtered), Argument);
 	StripQuotes(Filtered);
 	TrimString(Filtered);
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 35]
```diff
@@ -518,32 +1239,32 @@ public Action Command_SelfUnMute(int client, int args)
...
 	char Filtered[65];
 	strcopy(Filtered, sizeof(Filtered), Argument);
 	StripQuotes(Filtered);
 	TrimString(Filtered);
-
+	
 	if(UnMuteSpecial(client, Filtered))
 		return Plugin_Handled;
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 36]
```diff
@@ -518,32 +1239,32 @@ public Action Command_SelfUnMute(int client, int args)
...
 	if(UnMuteSpecial(client, Filtered))
 		return Plugin_Handled;
-
+	
 	char sTargetName[MAX_TARGET_LENGTH];
 	int aTargetList[MAXPLAYERS];
 	int TargetCount;
 	bool TnIsMl;
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 37]
```diff
@@ -557,44 +1278,90 @@ public Action Command_SelfUnMute(int client, int args)
 		ReplyToTargetError(client, TargetCount);
 		return Plugin_Handled;
 	}
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 38]
```diff
@@ -557,44 +1278,90 @@ public Action Command_SelfUnMute(int client, int args)
...
 	if(TargetCount == 1)
 	{
 		if(aTargetList[0] == client)
 		{
-			PrintToChat(client, "\x04[Self-Mute]\x01 Unmuting won't work either.");
+			CReplyToCommand(client, "%sUnmuting won't work either.", PLUGIN_PREFIX);
 			return Plugin_Handled;
 		}
-
+
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: remove spaces

File: addons/sourcemod/scripting/SelfMute.sp [hunk 39]
```diff
@@ -624,30 +1391,710 @@ public Action Command_CheckMutes(int client, int args)
...
+            else if(param2 == 1)
+            {
+				//if client chose to not save the selfmute
+
+                char buffer[128];
+                menu.GetItem(param2, buffer, sizeof(buffer));
+
+                if(buffer[0] == '@')
+                {
+					int SpecialMute = GetSpecialMutesFlags(buffer);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp:2063]: Identation

**Comment by maxime1907** [comment]
WOW, really nice work!! :smile:

File: addons/sourcemod/scripting/SelfMute.sp [hunk 40]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+		return;
+		
+	char sDriver[32];
+	g_hDB.Driver.GetIdentifier(sDriver, sizeof(sDriver));
+	if(StrEqual(sDriver, "mysql"))
+	{
+		Transaction T_mysqlTables = SQL_CreateTransaction();
+		
+		char sQuery0[1024];		
+		g_hDB.Format(sQuery0, sizeof(sQuery0), "CREATE TABLE IF NOT EXISTS `ClientsMute`("
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]:
```
CREATE TABLE IF NOT EXISTS `clients_mute`
```

File: addons/sourcemod/scripting/SelfMute.sp [hunk 41]
```diff
@@ -83,30 +140,503 @@ public void OnAllPluginsLoaded()
...
+												... "`client_steamid` varchar(32) NOT NULL," 
+												... "`target_name` varchar(1024) NOT NULL," 
+												... "`target_steamid` varchar(32) NOT NULL," 
+												... "PRIMARY KEY(`id`)," 
+												... "UNIQUE KEY(`target_steamid`))");
+																						
+		T_mysqlTables.AddQuery(sQuery0);
+	
+		char sQuery1[1024];	
+		g_hDB.Format(sQuery1, sizeof(sQuery1), "CREATE TABLE IF NOT EXISTS `GroupsMute`("
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]:
```
CREATE TABLE IF NOT EXISTS `groups_mute`
```

**Comment by maximee-leroyy** [comment]
WOW, really nice work!! :smile:

File: addons/sourcemod/scripting/SelfMute.sp [hunk 42]
```diff
@@ -83,30 +140,500 @@ public void OnAllPluginsLoaded()
...
+	char SteamID[32];	
+	if(!GetClientAuthId(client, AuthId_Steam2, SteamID, sizeof(SteamID)))
+		return Plugin_Stop;
+
+	g_bClientJustJoined[client] = true;
+	
+	Transaction T_ClientJoin = SQL_CreateTransaction();	
+				
+	char sQuery0[1024];
+	g_hDB.Format(sQuery0, sizeof(sQuery0), "SELECT `target_steamid` FROM `ClientsMute` WHERE client_steamid='%s'", SteamID);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: You forgot to change other references of tables new name

File: addons/sourcemod/scripting/SelfMute.sp [hunk 43]
```diff
@@ -624,30 +1386,710 @@ public Action Command_CheckMutes(int client, int args)
...
+			AddSavedGroupsToMenu(param1);
+		}
+	}
+	
+	return 0;
+}
+
+void DisplayUnSavedGroupsMuteMenu(int client)
+{
+	Menu menu = new Menu(MenuHandler_UnSavedGrouos);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: MenuHandler_UnSavedGroups

File: addons/sourcemod/scripting/SelfMute.sp [hunk 44]
```diff
@@ -624,30 +1386,710 @@ public Action Command_CheckMutes(int client, int args)
...
+	for(int i = 1; i <= 64; i++)
+	{
+		if(g_bClientUnSavedGroups[client][i])
+			count++;
+	}
+
+	return count;
+}
+
+public int MenuHandler_UnSavedGrouos(Menu menu, MenuAction action, int param1, int param2)
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/SelfMute.sp]: MenuHandler_UnSavedGroups

**Comment by Rushaway** [comment]
Here you go @maxime1907

**Comment by maximee-leroyy** [comment]
good job !!

**Pull request merged**